### ISSUE TITLE
Metadata: title template in layout.js doesn't work for page.js in the same level

### ISSUE THREAD

**Issue by joulev** [vercel/next.js#46859]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:42:11 PST 2023; root:xnu-8792.81.3~2/RELEASE_X86_64
Binaries:
  Node: 18.14.0
  npm: 9.3.1
  Yarn: 1.22.19
  pnpm: 7.29.0
Relevant packages:
  next: 13.2.4-canary.4
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Metadata (metadata, generateMetadata, next/head, head.js)

### Link to the code that reproduces this issue

https://github.com/joulev/debug/tree/nextjs-metadata-title-template-bug

### To Reproduce

* Clone the linked repo
* `next dev` or `next build`
* Check `/` and `/about`

Alternatively, check [the deployment](https://debug-git-nextjs-metadata-title-template-bug-joulev.vercel.app) for `/` and `/about`.

### Describe the Bug

`/about` title is "about | template" which is correct. But `/` title is "home" without the " | template" part, which is not correct.

### Expected Behavior

`/` title should be "home | template".

### Which browser are you using? (if relevant)

N/A

### How are you deploying your application? (if relevant)

N/A

**Comment by joulev** [comment]
After digging a bit I found this

https://github.com/vercel/next.js/blob/d59aa9655e7a46056a4e98c3a79d40a118f6157d/packages/next/src/lib/metadata/resolve-metadata.ts#L317-L325

So it does look like this is the desired behaviour... but may I ask why?

**Comment by joulev** [comment]
Ok this is indeed the desired behaviour (#45965), but I think I'll keep this issue open as I still don't understand the rationale behind this.

Also I think it's a good idea to update the documentation to reflect this, especially as the [Title templates](https://beta.nextjs.org/docs/api-reference/metadata#title-templates) section uses an example where `app/page.js` and a same-level `app/layout.js` are used, which makes it quite misleading.

**Comment by JesseKoldewijn** [comment]
Hey man, I'll have a look for ya if there might be a more feasible solution for the handling of the template and title

**Comment by Confuze** [comment]
Ran into this issue today, really weird. I added a comment to the documentation informing about this behavior. I have no clue why this would be desired.

**Comment by joulev** [comment]
Hmm, it gets even more interesting.

https://github.com/vercel/next.js/blob/14bfe87110fee29cda6dc30c29c0a2e336b069dd/packages/next/src/lib/metadata/resolve-metadata.test.ts#L32-L50

This shows that Next.js completely ignores all title templates except the nearest template. So you can't have like

```
Settings - owner/repo - GitHub
^          ^            ^
page.js    layout.js    (root) layout.js
```

**Comment by Confuze** [comment]
Huh, I think they'll just have to update this in a future version. I don't see how anybody could use the current behavior to their advantage.

**Comment by joulev** [comment]
I really want to make a PR since making one isn't hard, but these comments are really discouraging...

<img width="847" alt="image" src="https://user-images.githubusercontent.com/44609036/223469294-9d0c79e5-4043-4fa3-ad16-83dad38c55ad.png">

**Comment by huozhi** [comment]
This behavior is by designed, the layout only effects the pages under it, for the same layer page, it will be effected by the upper layer layout, or if it's root page you can specify one for it.

One example is:

| ur                         | title |
| ---------------- | ---- |
| `/`     | `"Brand"` |
| `/about` | `"About \| Brand"` |

The same level page might either blongs to the upper layouts or just be a special case

**Comment by JesseKoldewijn** [comment]
Maybe a toggle in the config would be a godd middleroad? To either choose for template title on all routes or just on the childs?

**Comment by joulev** [comment]
@JesseKoldewijn I think a config option would be an overkill in this case.

---

@huozhi I think in that case shouldn't it be like this? (It's literally the example in the documentation, actually.)

```ts
// layout.tsx
export const metadata = {
  title: {
    default: 'Brand',
    template: '%s | Brand',
  },
};

// page.tsx
export const metadata = {
  // just don't specify a title
};

// about/page.tsx
export const metadata = {
  title: "About";
}
```

Or we can also use `[ ]` for example to specify optional parts inside the template, so we can have `%s[ | ]Brand` for example?

Since `layout.tsx` also covers the same-level `page.tsx`, I found this design very confusing.

**Comment by huozhi** [comment]
@joulev If you don't specify title for root page then it will use the default title from root layout, it will render `"Brand"` if you don't specify any.

**Comment by JesseKoldewijn** [comment]
> @JesseKoldewijn I think a config option would be an overkill in this case.
> 
> ---
> 
> @huozhi I think in that case shouldn't it be like this? (It's literally the example in the documentation, actually.)
> 
> ```ts
> // layout.tsx
> export const metadata = {
>   title: {
>     default: 'Brand',
>     template: '%s | Brand',
>   },
> };
> 
> // page.tsx
> export const metadata = {
>   // just don't specify a title
> };
> 
> // about/page.tsx
> export const metadata = {
>   title: "About";
> }
> ```
> 
> Or we can also use `[ ]` for example to specify optional parts inside the template, so we can have `%s[ | ]Brand` for example?
> 
> Since `layout.tsx` also covers the same-level `page.tsx`, I found this design very confusing.

I btw mainly meant the config option as in an option in the metadata object btw. Just to clarify🤙

**Comment by joulev** [comment]
> @joulev If you don't specify title for root page then it will use the default title from root layout, it will render `"Brand"` if you don't specify any.

@huozhi Yeah so isn't that the desired behaviour in the table you showed above (`/` → `"Brand"` and `/about` → `"About | Brand"`), or am I missing something? I'm yet to find an example that would benefit from the current pattern and *not* benefit from the pattern I opened this issue for, so I still find the current pattern quite counter-intuitive.

**Comment by huozhi** [comment]
From your reproduction, if you remove the extra metadata export in root page, it will behave like you said. `(/ → "Brand" and /about → "About | Brand")`. I tested it, works as expected. Regarding to the original question I'll close this issue. But feel free to keep discussion.

**Comment by joulev** [comment]
Oh yeah, I was so focused on this issue that I completely missed that. Thanks.

**Comment by JesseKoldewijn** [comment]
> From your reproduction, if you remove the extra metadata export in root page, it will behave like you said. `(/ → "Brand" and /about → "About | Brand")`. I tested it, works as expected. Regarding to the original question I'll close this issue. But feel free to keep discussion.
> 

As far as I understood him he meant that it would be nicer to still have the ability to set the title + template for the root pathname "/". Which I kinda do get, but not sure if that's what he meant.

**Comment by Confuze** [comment]
> From your reproduction, if you remove the extra metadata export in root page, it will behave like you said. `(/ → "Brand" and /about → "About | Brand")`. I tested it, works as expected. Regarding to the original question I'll close this issue. But feel free to keep discussion.

It does work fine, but there's a problem when you want to specify a title for the root page.tsx. If you have a setup like this:
```
// layout.tsx
export const metadata = {
  title: {
    default: 'Brand',
    template: '%s | Brand',
  },
};

// page.tsx
export const metadata = {
  title: "Main page";
};

// about/page.tsx
export const metadata = {
  title: "About";
}
```

Then you'd get "Main page" instead of "Main page | Brand" as the title in the index of the website.

So as @joulev mentioned before I think it would be nicer to still have the ability to use the template in `app/page.tsx` (on the same level as the layout) since you could still have it use the default specified in `app/layout.tsx` by just not specifying the title in `app/page.tsx`

I think the issue should be opened because while I presume @joulev managed to get what they wanted, the original problem still remains unsolved.

**Comment by JesseKoldewijn** [comment]
fair

**Comment by Confuze** [comment]
Do you think you could re-open it? I don't have contributor permissions in here.

**Comment by JesseKoldewijn** [comment]
> 

Who are you asking?

**Comment by Confuze** [comment]
you?

**Comment by JesseKoldewijn** [comment]
Sorry man, I have no clue how to re-open an issue that was not made by myself. Didn't even know that contrib's could do that.

**Comment by Confuze** [comment]
It looks like only the author/vercel members can re-open issues, my bad.
@huozhi Could you please check out these latest comments and consider re-opening?

**Comment by joulev** [comment]
For the record I am the author and cannot reopen it so we need to wait for a maintainer to come and open.

**Comment by mjth** [comment]
I agree with @Confuze... and hope this issue is opened again @huozhi 

I was attempting to specify a title template in Root Layout, but it doesn't apply to the Root Page (and there is no way to do this). If this was designed in case you needed an escape hatch, you could either just not define the template on root, or override it (like you would for layouts further down the chain anyway...)

IMO it's also very confusing from a DX perspective...
The layout is designed to wrap the page on the same level... but the metadata only wraps children pages?

**Comment by huozhi** [comment]
> and there is no way to do this
@mjth as the example given above, `title.default` can handle the root page when the `title.template` is specified. Or you can give other title in root page it will also work. It's not "no way to do" like you said. The issue should be reopen as the behavior is expected and there're ways to solve it.

We also hear you loud on the request of more explanation and examples. We're working on the docs updates internally to clarify it in a better way. Please wait for the future docs updates with more clear example. 🙏

**Comment by Confuze** [comment]
@huozhi, I'm not sure if I understand correctly, but I don't think giving another title in the root page is a good sollution. If you want to have a behavior like this:
| Route url     | Rendered title  |
| ------------- | --------------  |
| /             | Home | Brand    |
| /about        | About | Brand   |
Then you'd have to manually write 'Home | Brand' in the metadata title in `app/page.tsx` like so:
```ts
// layout.tsx
export const metadata = {
  title: {
    default: 'Brand',
    template: '%s | Brand',
  },
};

// page.tsx
export const metadata = {
  title: "Home | Brand";
};

// about/page.tsx
export const metadata = {
  title: "About";
}
```
Which ultimately kind of defeats the purpose of title templates.

I don't see any way you could use the current behavior to your advantage and even if there is one I think it's safe to assume that there would be significantly more people wanting to use the title template in the root page (if they wanted to have just the default title on the root page, they could achieve it by just not specifying a title in the `app/page.tsx` metadata)

Also, as @mjth said, this is really bad DX-wise. Even if we introduce a change in the docs, it would still be really inconstant with the layout working on pages on the same level but the metadata not.

In my opinion the current behavior is really unintuitive and not really beneficial for anyone so I don't think a change in the documentation is the right way to approach this issue. And even if there would still be any need to use the current behavior, we could make it opt-in with a boolean, as @JesseKoldewijn. It could look something like this:
```ts
// layout.tsx
export const metadata = {
  title: {
    default: 'Brand',
    template: '%s | Brand',
  },
};

// page.tsx
export const metadata = {
  title: {
    content: "The home of brand",
    ignoreTemplates: true
  }
} // The final title would "The home of brand", without the " | Brand" suffix
```
That way you could also opt-out of templates in lower level files and the whole thing would be much more intuitive. (And as I mentioned before, you could still achieve the default "Brand" behavior by not specifying a title in a page's metadata)

**Comment by gnoff** [comment]
I'll try to articulate with an example why we have the current layering system in place. In the above examples it seems there is a lot of focus on how the root title it treated and we could change the inheritance of pages to make this case better but there is a use case that we could not support if we did that

Imagine we have the following pages structure
| path | title | notes |
| --- | --- | --- |
| / | Foo has the best products |
| /account | Account - Foo |
| /products | Products Home - Foo |
| /products/explore | Products - Foo | imagine this page or layout did not configure a specific title |
| /products/bar | Bar - A Foo Product |

how would we lay this out today
```tsx
// layout.js
export const metadata = {
  title: {
    default: 'Foo has the best products',
    template: '%s - Foo',
  },
};

// page.js
export const metadata = {
  // doesn't need title b/c the default suffices
};

// /account/page.js
export const metadata = {
  title: "Account" // uses template defined in root layout -> "Account - Foo"
};

// /products/layout.js
export const metadata = {
  title: {
    default: 'Products', // uses template defined in root layout -> "Products - Foo"
    template: '%s - A Foo Product ',
  },
};

// /products/page.js
export const metadata = {
  title: "Products Home" // uses template defined in root layout -> "Products Home - Foo"
};

// /products/explore/page.js
export const metadata = {
  // no title configured yet, uses nearest defined default
  // uses default title defined in /products -> "Products - Foo"
};

// /products/[productname]/layout.js
export const metadata = {
  title: capetalize(params.productname) // uses template defined in /products layout -> "Bar - A Foo Product"
};
```

This is a fairly contrived example but one thing it demonstrates is that if at a given folder level you need a different default title for the layout (for anything deeper that does not define a title itself) than the title for the page at that level and you want to use the layout defined at a higher level that would be impossible if Page was templatized by Layout within the same level.

The way that I think about this layering is:
> the template applies to titles defined in route segments deeper than it

In this regard `/page.js` is not deeper than `/layout.js` and so templates defined in the layout don't apply. I do understand that this can be confusing however if we changed this behavior then the above scenario would not be possible without manually repeating the template into the page specific title and defaults using absolutes.

So with this constraint in place the fact that the root `/page.js` does not use any template makes sense given there is no route segment "higher" than the root segment.

The main argument against this position that I can think of is that it's actually very rare and therefore even if we close off a possible use case it's better for the default behavior to be more intuitive. I'd like to know what folks in this thread think regarding their intuition vs the reasoning presented here. We will also need to be mindful that this thread is visited by folks who have had trouble with the default and there may be a large number of folks for whom this behavior is intuitive. I appreciate everyone for weighing in, it's important that we get this right and need your help :)

**Comment by JesseKoldewijn** [comment]
So would in this case maybe a override boolean be possible that defaupt to the current behaviour? This way you could have it both ways. This boolean as confuze layed out in a snippet could be included into the meta object as an optional override.

**Comment by gnoff** [comment]
@Confuze yeah there is a tradeoff. If you want the Page for a rouge segment to use the template that is defined for that same route segment then you'd either need to repeat it or extract to some external file

```tsx
// /title-helper.js
export const template = '%s - Brand';

// /layout.js
import { template } = 
export const metadata = {
    title: {
        default: "Brand",
        template,
    }
}

// /page.js
export const metadata {
    title: template.replace("%s", "Home"),
}
```

@JesseKoldewijn I'm not sure if I am misunderstanding the issue but I think what you'd want in that case is

```tsx
export const metadata = {
     title: {
       absolute: "The home of brand",
     }
}
```
using absolute opts the title out of using any templates

**Comment by gnoff** [comment]
@Confuze 
>Either way, I think the docs need to be updated so that the behavior is properly explained.

definitely, feedback already give to the docs team on how to clarify that templates are for deeper route segments and that a Page is not deeper than the Layout defined within the same route segment folder

**Comment by JesseKoldewijn** [comment]
> @Confuze yeah there is a tradeoff. If you want the Page for a rouge segment to use the template that is defined for that same route segment then you'd either need to repeat it or extract to some external file
> 
> ```tsx
> // /title-helper.js
> export const template = '%s - Brand';
> 
> // /layout.js
> import { template } = 
> export const metadata = {
>     title: {
>         default: "Brand",
>         template,
>     }
> }
> 
> // /page.js
> export const metadata {
>     title: template.replace("%s", "Home"),
> }
> ```
> 
> @JesseKoldewijn I'm not sure if I am misunderstanding the issue but I think what you'd want in that case is
> 
> ```tsx
> export const metadata = {
>      title: {
>        absolute: "The home of brand",
>      }
> }
> ```
> using absolute opts the title out of using any templates

Hey man, sorry for the confusion. I'm kinda doozy after a long day plus some stuff in my personal life. 

What I meant was something like discribed by confuze but sort of the other way around. Meaning the ability to override the current behaviour allowing the following title structure while still using the defined templates without having to duplicate or set the full title with template included as the defaupt.

Example:
```
// Override: true
Home | brand 
About | brand

// Override: false
Default | brand
About | brand
```

Ps. Sorry for the poor formatted example, I'm on my phone in the hospital for my grandpa.

**Comment by Confuze** [comment]
> @Confuze yeah there is a tradeoff. If you want the Page for a rouge segment to use the template that is defined for that same route segment then you'd either need to repeat it or extract to some external file
> 
> ```tsx
> // /title-helper.js
> export const template = '%s - Brand';
> 
> // /layout.js
> import { template } = 
> export const metadata = {
>     title: {
>         default: "Brand",
>         template,
>     }
> }
> 
> // /page.js
> export const metadata {
>     title: template.replace("%s", "Home"),
> }
> ```
> 
> @JesseKoldewijn I'm not sure if I am misunderstanding the issue but I think what you'd want in that case is
> 
> ```tsx
> export const metadata = {
>      title: {
>        absolute: "The home of brand",
>      }
> }
> ```
> using absolute opts the title out of using any templates

It's the first time I'm hearing about absolute and it seems to be the solution to ignoring templates but it doesn't actually give you the best of both worlds - I described the idea badly.

What I was trying to suggest was a way to let you have the have example you provided ("Products Home | Foo") and the one I provided ("Home | Foo") too, depending of what the developer wants.

So the actual way to go about this would be to add a boolean called something like useSameLevelTemplate (I know, it could use a better name) so that you could have both behaviors without manually inserting the templates. If you have trouble understanding what I mean, I can send a code example but I can only do it tomorrow as I'm writing this on my phone right before going to sleep.

This is an interesting problem though because while a solution like this does offer you the most options, I think it might be a bit tricky to implement it without making the whole template feature overly complicated and confusing

**Comment by Confuze** [comment]
> > @Confuze yeah there is a tradeoff. If you want the Page for a rouge segment to use the template that is defined for that same route segment then you'd either need to repeat it or extract to some external file
> > 
> > ```tsx
> > // /title-helper.js
> > export const template = '%s - Brand';
> > 
> > // /layout.js
> > import { template } = 
> > export const metadata = {
> >     title: {
> >         default: "Brand",
> >         template,
> >     }
> > }
> > 
> > // /page.js
> > export const metadata {
> >     title: template.replace("%s", "Home"),
> > }
> > ```
> > 
> > @JesseKoldewijn I'm not sure if I am misunderstanding the issue but I think what you'd want in that case is
> > 
> > ```tsx
> > export const metadata = {
> >      title: {
> >        absolute: "The home of brand",
> >      }
> > }
> > ```
> > using absolute opts the title out of using any templates
> 
> Hey man, sorry for the confusion. I'm kinda doozy after a long day plus some stuff in my personal life. 
> 
> What I meant was something like discribed by confuze but sort of the other way around. Meaning the ability to override the current behaviour allowing the following title structure while still using the defined templates without having to duplicate or set the full title with template included as the defaupt.
> 
> Example:
> ```
> // Override: true
> Home | brand 
> About | brand
> 
> // Override: false
> brand
> About | brand
> ```
> 
> Ps. Sorry for the poor formatted example, I'm on my phone in the hospital for my grandpa.

btw, this is exactly what I meant. @JesseKoldewijn posted this comment while I was writing mine lol.

**Comment by JesseKoldewijn** [comment]
> > @Confuze yeah there is a tradeoff. If you want the Page for a rouge segment to use the template that is defined for that same route segment then you'd either need to repeat it or extract to some external file
> > 
> > ```tsx
> > // /title-helper.js
> > export const template = '%s - Brand';
> > 
> > // /layout.js
> > import { template } = 
> > export const metadata = {
> >     title: {
> >         default: "Brand",
> >         template,
> >     }
> > }
> > 
> > // /page.js
> > export const metadata {
> >     title: template.replace("%s", "Home"),
> > }
> > ```
> > 
> > @JesseKoldewijn I'm not sure if I am misunderstanding the issue but I think what you'd want in that case is
> > 
> > ```tsx
> > export const metadata = {
> >      title: {
> >        absolute: "The home of brand",
> >      }
> > }
> > ```
> > using absolute opts the title out of using any templates
> 
> It's the first time I'm hearing about absolute and it seems to be the solution to ignoring templates but it doesn't actually give you the best of both worlds - I described the idea badly.
> 
> What I was trying to suggest was a way to let you have the have example you provided ("Products Home | Foo") and the one I provided ("Home | Foo") too, depending of what the developer wants.
> 
> So the actual way to go about this would be to add a boolean called something like useSameLevelTemplate (I know, it could use a better name) so that you could have both behaviors without manually inserting the templates. If you have trouble understanding what I mean, I can send a code example but I can only do it tomorrow as I'm writing this on my phone right before going to sleep.
> 
> This is an interesting problem though because while a solution like this does offer you the most options, I think it might be a bit tricky to implement it without making the whole template feature overly complicated and confusing

Sounds like we both sorta got the same idea. A boolean that allows the use of the current level template👌

**Comment by JesseKoldewijn** [comment]
Plus, by defaulting the title/template behaviour you allow the edge cases that require the current behaviour while still allowing bypassing the said behaviour.

Ps. My experience with dynamic meta data like titles etc come from using next-seo where it basically sort of behaves in the way of using the default if the title isn't set while still allowing overrides of this. (Depending on your own implementation of setting the template and/or overriding the template on certain directories/paths)

**Comment by gnoff** [comment]
@Confuze @JesseKoldewijn 

We try to keep our boolean config as limited as possible which is why the current set of configurable options is descriptive (absolute is a description of the absolute title rather than a flag like `skipTemplate: true`)

It may seem a little odd (b/c the title maybe seems like a minor reason to do this) but you can hook into the template behavior you want by structuring the directory slightly differently. route groups are the way to do this

```tsx
// /layout.js
export const metadata = {
  title: {
    default: "Brand",
    template: "%s | Brand",
  }
}

// if you use /page.js
export const metadata = {
  title: "Home | Brand" // title will be "Home | Brand" because the template is not used (same depth)
}

// try instead /(index)/page.js
export const metadata = {
  title: "Home" // title will be "Home | Brand" because the template is used (page is deeper than layout)
}

now your Page is in a "deeper" segment than the Layout. It's a little nuanced because earlier I said route-segment and really its based on folder hierarchy but usually these are the same thing. But in this case the page is part of a route-group which has no url representation but when the metadata resolves it is treated like it is a "child" of the Layout

that said, I personally wouldn't do this, I'd just use a shared helper to make the template uniform when I wanted this behavior

**Comment by gnoff** [comment]
@mjth 

>```tsx
>// /products/[productname]/page.js
>export const metadata = {
>  //This can be empty. Uses the closest template instead. -> 'Product Name - A Foo Product'
> //Since title has not been defined no %s is rendered.
>};
>```

I don't quite follow this part. this metadata doesn't declare a title, why does it use a template? Also wouldn't the computed title have an extra space if you did treat the title to be templatized as an empty string?
```
capatilise(params.productname) + ' %s- A Foo Product' -> "Product Name  - A Foo Product"
```
In trying to understand your examples though it made me realize that you might be thinking about how we "find" metadata for a page in a reverse order of sorts. My impression is you think about it as starting from the Page and looking up the Layout tree to find matching metadata whereas it actually starts with the root layout and computes a final value as it goes down. I'll try to clarify my mental model for the titles and see if this helps.

I think our Docs need to better clarify our intent here (or we need to make these properties longer) but here is how I think about the options

#### In a Layout
title defines default titles to use as well as templates to be applied to deeper title definitions
```
title (as string) -> apply nearest templateForChildTitles if any, this is the default title
title.absolute -> ignore any templates, this is the default title
title.default -> same as title (as string)
title.template -> set a new templateForChildTitles
```
these titles are always fallbacks for when you do not configure a title for your page. As we traverse each layout we keep track of the default title, augmenting it with templates as needed. We also keep track of whether a new template is to be used for deeper title definitions

#### In a Page
title defines the page title. If you don't provide a title the nearest default title is used
```
(no title) -> use the nearest default title
title (as string) -> apply nearest templateForChildTitles if any, this is the title used
title.absolute -> ignore any templates, this is the title used
```

It's important to note that a template never does anything on it's own. It always requires a title to augment (either a default title defined for a Layout or a title defined for a page)

The type structure could make this distinction more explicit
```tsx
// Layouts
{
  defaultTitle?: string | { absolute: string };
  titleTemplateForChildTitles?: string
}

// Pages
{
  title?: string | { absolute: string };
}
```
But generally we have a philosophy that it metadata is changing together it should be grouped together. This is why the template is embedded in the title definition and not a sibling property. We also try to communicate enough intent without using long property names.

Another thing that might help the mental model is to recognize that we don't start with the Page and bubble up to find the first matching title/template. We start at the root Layout and compute the title as we go deeper. If we never hit a Page that defines title then we just use the currently computed title. This is why we haven't really "skipped" the template. The template is just there to augment deeper definitions if one exists. sometimes one will not exist

I think we need to do some Docs updates to make this mental model show through clearer, but I do think our position is consistent and does allow for the greatest range of expression of titles.

If you want the behavior where the default title does not inherit the locally defined template you can do
```
{
  title: {
    absolute: "My Bespoke Default",
    template: "%s - The Template",
  }
}
```
And if you want the template to apply to the default you can easily do that
```
{
  title: {
    absolute: "Default With a Template - The Template",
    template: "%s - The Template",
  }
}
```

The most awkward case I can see is when you want your Page to use the template defined in your Layout but the same absolute trick works there too, it's just in a separate file.

**Comment by mjth** [comment]
Thanks @gnoff for your time in answering.  The description you've given is really clear.

In the first part, you are right around the title rendering - it probably should have had '' defined for it to render a title.

To be honest, the above was probably overkill - and I do understand that the title and template gets cascaded down until overwritten. The main point I was trying to make is that I believe it would make more sense for the template to define from the current level downwards, and that all the use cases would still be achievable- but with a better DX for the reasons I mentioned above (colocation of title/template, alignment of the template with the actual layout, etc).

I did have to reread the part about templateForChildTitles a few times to be honest, because I was still confused about the fact that the template didn't apply to the child **/products/explore/page.js.** but the default title did. 

I understand your logic, and it's probably just a matter of a difference of opinion on what the best developer experience would be at this stage. 

Thanks for the hard work.
